xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -969,6 +970,44 @@ void CGUIDialogPVRTimerSettings::RecordingGroupFiller(
     CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings::RecordingGroupFiller - No dialog");
 }
 
+void CGUIDialogPVRTimerSettings::MarginTimeFiller(
+  const CSetting *setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
+{
+  CGUIDialogPVRTimerSettings *pThis = static_cast<CGUIDialogPVRTimerSettings*>(data);
+  if (pThis)
+  {
+    list.clear();
+
+    // Get global settings values
+    CPVRSettings::MarginTimeFiller(setting, list, current, data);
+
+    if (setting->GetId() == SETTING_TMR_BEGIN_PRE)
+      current = pThis->m_iMarginStart;
+    else
+      current = pThis->m_iMarginEnd;
+
+    bool bFound = false;
+    for (const auto &entry : list)
+    {
+      if (entry.second == current)
+      {
+        bFound = true;
+        break;
+      }
+    }
+
+    // Insert current value if it is not in the list of predefined values.
+    // PVR backend may support more values than Kodi UI does.
+    if (!bFound)
+    {
+      list.push_back(
```

Comment:
From a UX point of view it would be a good thing to sort it. You can probably reuse the structure from https://github.com/kodi-pvr/pvr.hts/blob/master/src/tvheadend/ChannelTuningPredictor.h#L114 to accomplish it.